libsolidity/analysis/DocStringTagParser.cpp

```diff
@@ -47,6 +47,69 @@ bool DocStringTagParser::parseDocStrings(SourceUnit const& _sourceUnit)
 	return errorWatcher.ok();
 }
 
+bool DocStringTagParser::validateDocStringsUsingTypes(SourceUnit const& _sourceUnit)
+{
+	auto errorWatcher = m_errorReporter.errorWatcher();
+
+	SimpleASTVisitor visitReturns(
+		[](ASTNode const&) { return true; },
+		[&](ASTNode const& _node)
+		{
+			if (auto const* annotation = dynamic_cast<StructurallyDocumentedAnnotation const*>(&_node.annotation()))
+			{
+				auto const& documentationNode = dynamic_cast<StructurallyDocumented const&>(_node);
+
+				size_t returnTagsVisited = 0;
+
+				for (auto const& [tagName, tagValue]: annotation->docTags)
+					if (tagName == "return")
+					{
+						returnTagsVisited++;
+						vector<string> returnParameterNames;
+
+						if (auto const* varDecl = dynamic_cast<VariableDeclaration const*>(&_node))
+						{
+							if (!varDecl->isPublic())
+								continue;
```

Comment:
Kind of.
Previously the function was called using
```
		if (_variable.isPublic())
			parseDocStrings(_variable, _variable.annotation(), {"dev", "notice", "return", "inheritdoc"}, "public state variables");

```
So it would only see public state variables and as you said, it also has a more limited set of valid tags.

In the new function it is just called from compilerstack directly and goes over all already parsed tags.